core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -2204,11 +2209,22 @@ private void processTxFromBestChain(Transaction tx, boolean forceAddToPool) thro
             addWalletTransaction(Pool.SPENT, tx);
         }
 
-        // Kill txns in conflict with this tx
-        Set<Transaction> doubleSpendTxns = findDoubleSpendsAgainst(tx, pending);
-        if (!doubleSpendTxns.isEmpty()) {
-            // no need to addTransactionsDependingOn(doubleSpendTxns) because killTxns() already kills dependencies;
-            killTxns(doubleSpendTxns, tx);
+        if(!findDoubleSpendsAgainst(tx, transactions).isEmpty()) {
```

Comment:
Generally peers could sent any transaction, since bloom filters are designed to have false positives (or the wallet may not use bloom filters). The more robust we are against receiving unexpected transactions, the better.